### ISSUE TITLE
Accessory with multiple Services 

### ISSUE THREAD

**Issue by burgstaller** [homebridge/homebridge#3281]
### Current Situation

I based my plugin the available [example plugin](https://github.com/homebridge/homebridge-plugin-template) template. 
I have implemented an irrigation system with several valves to control each zone and a switch to turn all zones on/off.

**My questions is:** if an accessory "contains" several services, how is it decided which service is displayed as the "main" service in the homekit app and which services are "contained" in the main service? Is it just decided by the order the `accessory.addService(..)` is called? (e.g. the first service is the "main" service that is displayed top level in the home app?)

From my above example implementation:
I first added a new irrigation system service, then 3 valves and then 1 switch.
In the home app, this is displayed as a irrigation system that contains the other 4 services inside it. (To be clear: this is what i actually want, however I dont fully understand how/why this works)

My setup:
```
this.service = this.accessory.getService(this.platform.Service.IrrigationSystem) || this.accessory.addService(this.platform.Service.IrrigationSystem);
..
const valveService1 = this.accessory.getService(valveName) || this.accessory.addService(this.platform.Service.Valve, valveName, valveId);
const valveService2 = this.accessory.getService(valveName2) || this.accessory.addService(this.platform.Service.Valve, valveName2, valveId2);
const valveService3 = this.accessory.getService(valveName3) || this.accessory.addService(this.platform.Service.Valve, valveName3, valveId3);
...
const switchService = this.accessory.getService(this.platform.Service.Switch) || this.accessory.addService(this.platform.Service.Switch);
```
 How is it decided how the above accessory is displayed in the home app?

**Additionally:** 
Since I upgraded to iOS 16 the 3 Valves are not displayed with the original/correct name anymore but with the display name of the irrigation system they are contained in. (on my old phone and also on the mac app the names are displayed correctly)
Can someone confirm that this is a bug? 
### Logs

```markdown
-
```

### Configuration

```json
-
```

### Environment

- Homebridge: 1.6.0

### Process Supervisor

hb-service

**Comment by ebaauw** [comment]
> How is it decided how the above accessory is displayed in the home app?

HomeKit has the concept of a primary service, which you can set through `Service.setPrimaryService()`.  It also has the concept of a linked service.  I think these are only hints to HomeKit apps how to display the service, not sure if they do anything else.  We did see in a TV accessory that some services need to be linked.

Note that services are not “contained” in other services.  Services are contained in an accessory.  Apple’s Home app lies through its teeth, calling both accessories and services “Accessory”.  The _Show as Single Tile_ / _Show as Separate Tiles_ setting determines whether each service is shown as a separate tile.  Siri acts on services, not on accessories.

> Can someone confirm that this is a bug?

I’m afraid it’s an undocumented feature.  Pre-iOS 16, accessory names and service names would be maintained by Homekit, invisible to the accessory (i.c. Homebridge).  The would be initialised from the accessory name and service names, as exposed by the accessory.  In iOS 16, it looks like the service names are initialised from the accessory name as well.
However, as discussed on Discord, it seems iOS 16 now accepts _Configured Name_ on each service.  It syncs the service name with the value of this characteristic, finally allowing a plugin to see the service name when changed from HomeKit, or even change the service name.

**Comment by burgstaller** [comment]
Thanks for the info, that really cleared things up!

I have played arround with `setPrimaryService` and `addLinkedService` however nothing is changed in the way how the home app displays the services. (which is fine, I just wanted to understand how this works in theory)

Also the `ConfiguredName` Characteristic works perfectly, thanks!

I just discovered something new with the home app unter iOS 16:
when i add the homebridge in the new home app, the irrigation system now is displayed differently:
homekit seems to understand that each valve is a "zone" in the irrigation system (which is cool) however the switch (which i use to start a program that starts each individual zone) is not displayed anymore as a separate tile (only the valves). The switch is now hidden behind a button ("Geräte" in german, i assume this is accessories oder devices in the english app)
is there a way to control how a linked service is displayed?

many thanks!!
